queries.sql

```diff
@@ -28,4 +28,88 @@ WHERE name <> 'Gabumon';
 
 SELECT *
 FROM animals
-WHERE weight_kg >= 10.4 AND weight_kg <= 17.3;
\ No newline at end of file
+WHERE weight_kg >= 10.4 AND weight_kg <= 17.3;
+
+
+BEGIN TRANSACTION;
+
+UPDATE animals
+SET species = 'unspecified';
+
+
+UPDATE animals
+SET species = 'digimon'
+WHERE name LIKE '%mon';
+
+UPDATE animals
+SET species = 'pokemon'
+WHERE species IS `unspecified`;
+
+SELECT * FROM animals;
+
+COMMIT;
+
+SELECT * FROM animals;
+
+
+
+BEGIN TRANSACTION;
+
+DELETE FROM animals;
+
+SELECT * FROM animals;
+
+ROLLBACK;
+
+SELECT * FROM animals;
+
+
+BEGIN TRANSACTION;
+
+DELETE FROM animals
+WHERE date_of_birth > '2021-01-01';
+
+SAVEPOINT deleted_data_with_date_2021_01_01;
+
+UPDATE animals
+SET weight_kg = weight_kg * -1;
+
+ROLLBACK TO SAVEPOINT deleted_data_with_date_2021_01_01;
+
+UPDATE animals
+SET weight_kg = weight_kg * -1
+WHERE weight_kg < 0;
+
+ROLLBACK;
+
+SELECT * FROM animals;
+
+COMMIT;
+
+
+SELECT COUNT(*) AS total_animals
+FROM animals;
+
+SELECT COUNT(*) AS non_escaping_animals
+FROM animals
+WHERE escape_attempts = 0;
+
+SELECT AVG(weight_kg) AS average_weight
+FROM animals;
+
+SELECT neutered, COUNT(*) AS escape_count
+FROM animals
+WHERE escape_attempts > 0
+GROUP BY neutered
+ORDER BY escape_count DESC
+LIMIT 1;
+
+SELECT animals, MIN(weight_kg) AS min_weight, MAX(weight_kg) AS max_weight
+FROM animals
+GROUP BY animals;
+
+SELECT animals, AVG(escape_attempts) AS average_escape_attempts
+FROM animals
+WHERE date_of_birth BETWEEN '1990-01-01' AND '2000-12-31'
+GROUP BY animals;
```

Comment:
- [ ] As the one above, there should also be a 'group by species' statement as per the question: 'What is the average number of escape attempts per animal type of those born between 1990 and 2000?'. Kindly fix this and update the screenshot for your applied changes to queries.sql.